packages/ember-data/lib/system/model/states.js

```diff
@@ -343,7 +345,7 @@ var DirtyState = {
 
     rolledBack: function(record) {
       get(record, 'errors').clear();
-      record.triggerLater('ready');
+      record.transitionTo('loaded.saved');
```

Comment:
Actually it looks like the above test fails on current `master` as well. I will investigate and suspect the test setup is not correct.

But still, getting the state of the new, invalid person after it is rolled back via `person.currentState.stateName` is `root.loaded.saved`, which should be `root.deleted.saved`...